### PR TITLE
added view template support for get apis

### PR DESCRIPTION
**Pull request by tejash-jl**
- added view template support for get apis
- added remove paths provider for view templates

### CODE REVIEW

**Review by varadeth — APPROVED** [review]

**Comment by tejash-jl** [comment]
https://github.com/Sunbird-RC/community/issues/362

**Comment by tejash-jl** [comment]
- [ ] document view templates

**Comment by tejash-jl** [comment]
by default, it should not remove non public fields while using GET API if the user is owner or schema is anonymous

File: java/registry/src/main/java/dev/sunbirdrc/registry/service/NativeSearchService.java [hunk 1]
```diff
@@ -140,21 +141,27 @@ public JsonNode search(JsonNode inputQueryNode) throws IOException {
...
 	}
 
-	private void removeNonPublicFields(ArrayNode result, SearchQuery searchQuery, ObjectNode shardResult) throws Exception {
+	private ArrayNode removeNonPublicFields(SearchQuery searchQuery, ObjectNode shardResult) throws Exception {
+		ArrayNode result = JsonNodeFactory.instance.arrayNode();
 		for(String entityType: searchQuery.getEntityTypes()) {
 			ArrayNode arrayNode = (ArrayNode) shardResult.get(entityType);
-			for(JsonNode node : arrayNode) {
-				result.add(JSONUtil.removeNodesByPath(node, definitionsManager.getExcludingFieldsForEntity(entityType)));
+			if (removeNonPublicFieldsForNativeSearch) {
```
Comment (tejash-jl) [inline java/registry/src/main/java/dev/sunbirdrc/registry/service/NativeSearchService.java:152]: by default, it should not remove non public fields while using GET API if the user is owner or schema is anonymous

File: java/view-templates/src/main/java/dev/sunbirdrc/provider/RemovePathFunctionProvider.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import dev.sunbirdrc.views.IViewFunctionProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class RemovePathFunctionProvider implements IViewFunctionProvider<JsonNode> {
```
Comment (varadeth) [inline java/view-templates/src/main/java/dev/sunbirdrc/provider/RemovePathFunctionProvider.java:14]: this class can be moved to org/example/provider

**Comment by tejash-jl** [comment]
add e2e tests

**Pull request merged**